### PR TITLE
BUGZ-511: fix crash in MyAvatar::updateStaleAvatarEntityBlobs()

### PR DESCRIPTION
**Pull request by AndrewMeadows**
https://highfidelity.atlassian.net/browse/BUGZ-511

This PR fixes a rare crash in `MyAvatar::updateStaleAvatarEntityBlobs()`.

### CODE REVIEW

File: interface/src/avatar/MyAvatar.cpp [hunk 1]
```diff
@@ -2488,12 +2488,12 @@ QVariantList MyAvatar::getAvatarEntitiesVariant() {
...
             avatarEntityData["id"] = entityID;
             EntityItemProperties entityProperties = entity->getProperties(desiredProperties);
-            QScriptValue scriptProperties;
             {
                 std::lock_guard<std::mutex> guard(_scriptEngineLock);
+                QScriptValue scriptProperties;
                 scriptProperties = EntityItemPropertiesToScriptValue(_scriptEngine, entityProperties);
+                avatarEntityData["properties"] = scriptProperties.toVariant();
             }
-            avatarEntityData["properties"] = scriptProperties.toVariant();
```
Comment (AndrewMeadows) [inline interface/src/avatar/MyAvatar.cpp]: There was a previous crash (BUGZ-420) with fix (PR-15654) where a last-minute suggestion to pull scriptProperties out of the context block may have introduced this bug.

The problem here was: calling `QScriptValue::toVariant()` will reach back into the `QScriptEngine` instance used to fill the `QScriptValue`.  Apparently the `QScriptValue` keeps a back pointer to the engine and can call secret (private) methods on it.  This is made evident by looking carefully at the callstack of one of the threads:

```
    [7] interface.app/Contents/Frameworks/QtScript.framework/Versions/5/QtScript!QScriptEnginePrivate::variantMapFromObject(QTJSC::ExecState*, QTJSC::JSObject*)
    [8] interface.app/Contents/Frameworks/QtScript.framework/Versions/5/QtScript!QScriptEnginePrivate::toVariant(QTJSC::ExecState*, QTJSC::JSValue) 
    [9] interface.app/Contents/Frameworks/QtScript.framework/Versions/5/QtScript!QScriptValue::toVariant() 
    [10] interface.app/Contents/MacOS/interface!MyAvatar::getAvatarEntitiesVariant()
```

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19880/HighFidelity-Beta-PR15726-dbce976.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19880/HighFidelity-Quest-Beta-PR15726-dbce976.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19880/HighFidelity-Beta-PR15726-dbce976.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19880/HighFidelity-Beta-Interface-PR15726-dbce976.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19880/HighFidelity-Beta-PR15726-dbce976.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19880/HighFidelity-Beta-Interface-PR15726-dbce976.exe

**Pull request merged**